torch/jit/__init__.py

```diff
@@ -261,14 +262,15 @@ def __init__(self, inner, nderivs=0):
 
     def forward(self, *args):
         global _tracing
-        in_vars = _flatten(args)
+        in_vars, in_desc = _flatten(args)
         # NOTE: use full state, because we need it for BatchNorm export
         # This differs from the compiler path, which doesn't support it at the moment.
         module_state = list(self.state_dict(keep_vars=True).values())
-        trace = torch._C._tracer_enter(in_vars + module_state, self.nderivs)
+        trace, all_trace_inputs = torch._C._tracer_enter(in_vars + module_state, self.nderivs)
         _tracing = True
-        out = self.inner(*args)
-        out_vars = _flatten(out)
+        trace_inputs = _unflatten(all_trace_inputs[:len(in_vars)], in_desc)
```

Comment:
Do this more safely. What is happening here is you are creating views for the module parameters if they alias, and then throwing them all out because you don't have a convenient way of swizzling them with the view'ed versions. Much *better* to detect if there are duplicate module parameters, and complain to the user immediately, rather than silently have the old behavior.

Reply:
The reason why I'm not updating the module parameters is not only because there's no easy way to do this, but also because it's not needed - the only situation when we have to overwrite inputs is when they are repeated, but I think it's reasonable to assume that they are unique (they are in this case)

Reply:
Ah it's a module dict, so we need to deduplicate it ourselves... Not sure how will this interact with ONNX